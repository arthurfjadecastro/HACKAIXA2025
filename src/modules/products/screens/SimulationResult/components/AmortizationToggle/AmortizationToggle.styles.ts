import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  amortizationToggle: {
    marginTop: 12, // Reduzido de 16 para 12
    paddingTop: 12, // Reduzido de 16 para 12
    borderTopWidth: 1,
    borderTopColor: '#F0F0F0',
  },

  toggleLabel: {
    fontSize: 13, // Reduzido de 14 para 13
    fontWeight: '600',
    color: '#333333',
    marginBottom: 8, // Reduzido de 12 para 8
  },

  toggleButtons: {
    flexDirection: 'row',
    backgroundColor: '#F8F9FA',
    borderRadius: 8,
    padding: 3, // Reduzido de 4 para 3
  },

  toggleButton: {
    flex: 1,
    paddingVertical: 10, // Reduzido de 12 para 10
    paddingHorizontal: 14, // Reduzido de 16 para 14
    borderRadius: 6,
    alignItems: 'center',
  },

  toggleButtonActive: {
    backgroundColor: '#005CA9',
  },

  toggleButtonInactive: {
    backgroundColor: 'transparent',
  },

  toggleButtonText: {
    fontSize: 13, // Reduzido de 14 para 13
    fontWeight: '600',
  },

  toggleButtonTextActive: {
    color: '#FFFFFF',
  },

  toggleButtonTextInactive: {
    color: '#666666',
  },
});
