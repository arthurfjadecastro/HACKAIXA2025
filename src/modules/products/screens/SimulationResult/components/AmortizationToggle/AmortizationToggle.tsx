import React from 'react';
import { View, TouchableOpacity, StyleSheet } from 'react-native';
import { Text } from '@/design-system/components';
import { styles } from './AmortizationToggle.styles';

interface AmortizationToggleProps {
  amortizationType: 'PRICE' | 'SAC';
  onToggle: (type: 'PRICE' | 'SAC') => void;
}

export const AmortizationToggle: React.FC<AmortizationToggleProps> = ({
  amortizationType,
  onToggle,
}) => {
  return (
    <View style={styles.amortizationToggle}>
      <Text style={styles.toggleLabel}>Comparar sistemas:</Text>
      <View style={styles.toggleButtons}>
        <TouchableOpacity
          style={[
            styles.toggleButton,
            amortizationType === 'PRICE' 
              ? styles.toggleButtonActive 
              : styles.toggleButtonInactive
          ]}
          onPress={() => onToggle('PRICE')}
        >
          <Text style={StyleSheet.flatten([
            styles.toggleButtonText,
            amortizationType === 'PRICE' 
              ? styles.toggleButtonTextActive 
              : styles.toggleButtonTextInactive
          ])}>
            PRICE
          </Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={[
            styles.toggleButton,
            amortizationType === 'SAC' 
              ? styles.toggleButtonActive 
              : styles.toggleButtonInactive
          ]}
          onPress={() => onToggle('SAC')}
        >
          <Text style={StyleSheet.flatten([
            styles.toggleButtonText,
            amortizationType === 'SAC' 
              ? styles.toggleButtonTextActive 
              : styles.toggleButtonTextInactive
          ])}>
            SAC
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};
