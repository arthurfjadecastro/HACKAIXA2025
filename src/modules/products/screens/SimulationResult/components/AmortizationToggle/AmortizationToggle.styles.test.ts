import { styles } from './AmortizationToggle.styles';

describe('AmortizationToggle.styles', () => {
  it('should export styles object', () => {
    expect(styles).toBeDefined();
    expect(typeof styles).toBe('object');
  });

  it('should have amortizationToggle style properties', () => {
    expect(styles.amortizationToggle).toBeDefined();
    expect(styles.amortizationToggle.marginTop).toBe(12);
  });

  it('should have toggleLabel style properties', () => {
    expect(styles.toggleLabel).toBeDefined();
    expect(styles.toggleLabel.fontSize).toBe(13);
    expect(styles.toggleLabel.fontWeight).toBe('600');
  });

  it('should have consistent style structure', () => {
    expect(styles).toMatchSnapshot();
  });
});
