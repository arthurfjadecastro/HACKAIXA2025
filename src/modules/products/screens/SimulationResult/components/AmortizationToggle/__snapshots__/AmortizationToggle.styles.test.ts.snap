// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmortizationToggle.styles should have consistent style structure 1`] = `
{
  "amortizationToggle": {
    "borderTopColor": "#F0F0F0",
    "borderTopWidth": 1,
    "marginTop": 12,
    "paddingTop": 12,
  },
  "toggleButton": {
    "alignItems": "center",
    "borderRadius": 6,
    "flex": 1,
    "paddingHorizontal": 14,
    "paddingVertical": 10,
  },
  "toggleButtonActive": {
    "backgroundColor": "#005CA9",
  },
  "toggleButtonInactive": {
    "backgroundColor": "transparent",
  },
  "toggleButtonText": {
    "fontSize": 13,
    "fontWeight": "600",
  },
  "toggleButtonTextActive": {
    "color": "#FFFFFF",
  },
  "toggleButtonTextInactive": {
    "color": "#666666",
  },
  "toggleButtons": {
    "backgroundColor": "#F8F9FA",
    "borderRadius": 8,
    "flexDirection": "row",
    "padding": 3,
  },
  "toggleLabel": {
    "color": "#333333",
    "fontSize": 13,
    "fontWeight": "600",
    "marginBottom": 8,
  },
}
`;
